classes/Console/Command/AdminDemoteCommand.php

```diff
@@ -13,44 +13,80 @@
 
 namespace OpenCFP\Console\Command;
 
-use OpenCFP\Console\BaseCommand;
+use OpenCFP\Domain\Services;
+use OpenCFP\Infrastructure\Auth;
+use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
 
-class AdminDemoteCommand extends BaseCommand
+final class AdminDemoteCommand extends Command
 {
+    /**
+     * @var Services\AccountManagement
+     */
+    private $accountManagement;
+
+    public function __construct(Services\AccountManagement $accountManagement)
+    {
+        parent::__construct('admin:demote');
+
+        $this->accountManagement = $accountManagement;
+    }
+
     protected function configure()
     {
         $this
-            ->setName('admin:demote')
+            ->setDescription('Demote an existing user from admin')
             ->setDefinition([
-                new InputArgument('email', InputArgument::REQUIRED, 'Email address of user to demote'),
+                new InputArgument('email', InputArgument::REQUIRED, 'Email address of user to demote from admin'),
             ])
-            ->setDescription('Demote an existing user from being an admin')
             ->setHelp(
                 <<<EOF
-The <info>%command.name%</info> command removes a user from the admin group for a given environment:
+The <info>%command.name%</info> command demotes a user from the admin group for a given environment:
 
 <info>php %command.full_name% speaker@opencfp.org --env=production</info>
 <info>php %command.full_name% speaker@opencfp.org --env=development</info>
 EOF
-);
+            );
     }
 
     public function execute(InputInterface $input, OutputInterface $output)
     {
-        return $this->demote($input, $output, 'Admin');
-    }
+        $email = $input->getArgument('email');
 
-    /**
-     * Method used to inject a OpenCFP\Application object into the command
-     * for testing purposes
-     *
-     * @param $app \OpenCFP\Application
-     */
-    public function setApp($app)
-    {
-        $this->app = $app;
+        $io = new SymfonyStyle(
+            $input,
+            $output
+        );
+
+        $io->title('OpenCFP');
+
+        $io->section(\sprintf(
+            'Demoting account with email "%s" from "Admin"',
+            $email
+        ));
+
+        try {
+            $this->accountManagement->findByLogin($email);
+        } catch (Auth\UserNotFoundException $exception) {
+            $io->error(\sprintf(
+                'Could not find account with email "%s".',
+                $email
+            ));
+
+            return 1;
+        }
+
+        $this->accountManagement->demoteFrom(
```

Comment:
What about adding a check to see if the user is actually an admin before trying to demote the account. And if the account is not an admin, telling the user so, what do you think?

Reply:
@BackEndTea 

I have been thinking about that ever since I started decoupling these commands, however, since this affects all of them, I would like to do it all in one go. What do you think?

Reply:
In that case it would make more sense to do that all in one go.